Targets/Mashow/Sources/DrinkSelection/View/DrinkSelectionViewController.swift

```diff
@@ -0,0 +1,287 @@
+        blurEffectView.layer.cornerRadius = 13
+        blurEffectView.layer.masksToBounds = true
+        blurEffectView.isUserInteractionEnabled = false
+        blurEffectView.translatesAutoresizingMaskIntoConstraints = false
+        button.insertSubview(blurEffectView, at: 0)
+        
+        NSLayoutConstraint.activate([
+            blurEffectView.leadingAnchor.constraint(equalTo: button.leadingAnchor),
+            blurEffectView.trailingAnchor.constraint(equalTo: button.trailingAnchor),
+            blurEffectView.topAnchor.constraint(equalTo: button.topAnchor),
+            blurEffectView.bottomAnchor.constraint(equalTo: button.bottomAnchor)
+        ])
+        return button
+    }()
+    
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        bind()
+        setupNavigationBar()
+        setupLayouts()
+        setupHandlers()
+    }
+}
+
+private extension DrinkSelectionViewController {
+    
+    private func bind() {
+        viewModel.state.currentType
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] currentType in
+                guard let self = self else { return }
+                setupBackground(currentType.rawValue)
+            }
+            .store(in: &cancellables)
+        
+        viewModel.state.addedTypesPublisher
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] addedTypes in
+                guard let self = self else { return }
+                if let tag = self.removedButtonTag {
+                    // If type removed
+                    UIView.transition(with: addedTypesStackView, duration: 0.5, options: .transitionCrossDissolve) {
+                        self.addedDrinkTypeButtons[tag]!.removeFromSuperview()
+                        self.addedDrinkTypeButtons[tag] = nil
+                        self.removedButtonTag = nil
+                    }
+                } else {
+                    guard !addedTypes.isEmpty else { return }
+                    // If type added
+                    let newButton = self.makeAddedTypeButton()
+                    newButton.addTarget(self, action: #selector(didTapToRemoveType), for: .touchUpInside)
+                    self.addedDrinkTypeButtons[viewModel.state.currentType.value.tag] = newButton
+                    UIView.transition(with: addedTypesStackView, duration: 0.5, options: .transitionCrossDissolve) {
+                        self.addedTypesStackView.addArrangedSubview(newButton)
+                    }
+                }
+            }
+            .store(in: &cancellables)
+    }
+    
+    @objc private func didTapToRemoveType(_ sender: UIButton) {
+        removedButtonTag = sender.tag
+        guard let index = drinkTypeList.firstIndex(where: { $0.tag == removedButtonTag }) else { return }
+        viewModel.removeType(drinkTypeList[index])
+    }
+    
+    private func makeAddedTypeButton() -> UIButton {
```

Comment:
```suggestion
    private func makeAddedTypeButton(for type: DrinkType) -> UIButton {
```
이렇게해서 `currentType`을 밖에서 받으면 함수가 순수해져서 보기 좋을 것 같아용